pkg/cloudfabric-controller/client_builder.go

```diff
@@ -58,8 +58,8 @@ type SimpleControllerClientBuilder struct {
 }
 
 func (b SimpleControllerClientBuilder) Config(name string) (*restclient.Config, error) {
-	clientConfig := *b.ClientConfig
-	return restclient.AddUserAgent(&clientConfig, name), nil
+	clientConfig := restclient.CopyConfigs(b.ClientConfig)
```

Comment:
will this only be used by the insecure mode ? if not, likely it will end up some unnecessary copies of config  objects.

Reply:
Actually, SimpleControllerClientBuilder is a struct that will be used by insecure mode only. Secure mode uses SAControllerClientBuilder. See func Run() in cmd/kube-controller-manager/app/controllermanager.go.

Reply:
good to know.